git/git.go

```diff
@@ -445,6 +420,12 @@ func (git *Git) Add(files ...string) error {
 	return err
 }
 
+// Clone will clone the given repo inside the given dir.
+func (git *Git) Clone(repoURL string, dir string) error {
```

Comment:
by current design, the porcelain commands are guarded by:

```
    if !git.config.AllowPorcelain {
		return fmt.Errorf("Init: %w", ErrDenyPorcelain)
	}
```
I think you should do the same here and then for the case of the vendor command we created the wrapper explicitly allowing porcelain.